### PR TITLE
[qscintilla] Update to 2.11.4

### PR DESCRIPTION
**Pull request by longhuan2018**
update qscintilla version from 2.10 to 2.11.4
change Homepage from "https://sourceforge.net/projects/pyqt" to "https://www.riverbankcomputing.com/software/qscintilla"

### CODE REVIEW

**Comment by longhuan2018** [comment]
I don't know why it went wrong, I checked the log of CL and found that compiling qscintilla was successful

File: ports/qscintilla/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: qscintilla
-Version: 2.10-11
-Homepage: https://sourceforge.net/projects/pyqt
+Version: 2.11.4
+Homepage: https://www.riverbankcomputing.com/software/qscintilla
 Description: QScintilla is a port to Qt of the Scintilla editing component. Features syntax highlighting, code-completion and much more (Barebone build without python bindings (missing dependeny PyQt) and without QtDesigner plugin)
 Build-Depends: qt5-base, qt5-macextras (osx), qt5-winextras (windows)
```
Comment (NancyLi1013) [inline ports/qscintilla/CONTROL]: It seems that this port currently doesn't support `linux` and `osx`.
Could you please add `Supports `filed here?
`Supports: windows`

File: ports/qscintilla/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,7 @@
-include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/qscintilla/portfile.cmake]: Could you please also add` vcpkg_fail_port_install(ON_TARGET "linux" "osx")` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @longhuan2018 
Thanks for this PR.
There is something wrong with CI test and we have been trying to fix the regressions in this PR #10508.
Once fixed, we will re run this PR.

File: ports/qscintilla/portfile.cmake (hunk 2 above)
Comment (longhuan2018) [reply ports/qscintilla/portfile.cmake]: Ok,  i will try.

File: ports/qscintilla/CONTROL (hunk 1 above)
Comment (longhuan2018) [reply ports/qscintilla/CONTROL]: Ok,  i will try.

**Comment by longhuan2018** [comment]
> Hi @longhuan2018
> Thanks for this PR.
> There is something wrong with CI test and we have been trying to fix the regressions in this PR #10508.
> Once fixed, we will re run this PR.

OK, I just modified it according to your prompt, but CI still fails. I'm sure QScintilla supports linux and osx, so I will revert these changes and wait for the CI problem to be solved.

**Comment by NancyLi1013** [comment]
Thanks for your update.

I noticed it was set as fail on linux and osx in ci.baseline.txt.
If you can confirm this port supports linux and osx, there is no need to add the prompt I suggested.
And we should remove the following codes from c.baseline.txt.
```
qscintilla:x64-linux=fail
qscintilla:x64-osx=fail
```
But I saw the CI test results, only linux is passing. So please help confirm if it supports on osx.
Of course,you don't need to update this now until the CI problem is fixed.

**Comment by NancyLi1013** [comment]
You don't need to care about other regressions that are not related with this PR.
Let's just wait for the CI problem is fixed.
Once fixed, I will re run this PR.

**Comment by NancyLi1013** [comment]
It seems that it has passed on Linux platform.
`##[error] PASSING, REMOVE FROM FAIL LIST: qscintilla:x64-linux (/ci/myagent/_work/3/s/scripts/ci.baseline.txt)`
Please update ci.baseline.txt.

Also, please add `vcpkg_fail_port_install(ON_TARGET "osx")` to top of `portfile.cmake`
add `Supports: !osx` to `CONTROL `file.

**Comment by longhuan2018** [comment]
> It seems that it has passed on Linux platform.
> `##[error] PASSING, REMOVE FROM FAIL LIST: qscintilla:x64-linux (/ci/myagent/_work/3/s/scripts/ci.baseline.txt)`
> Please update ci.baseline.txt.
> 
> Also, please add `vcpkg_fail_port_install(ON_TARGET "osx")` to top of `portfile.cmake`
> add `Supports: !osx` to `CONTROL `file.

OK, I will modify it according to your opinion. In addition, I think qscintilla supports osx. How can I make CI compile it?

**Comment by NancyLi1013** [comment]
If it supports osx, you can remove `qscintilla:x64-osx` from ci.baseline.txt.
Then try to fix the regressions.
It might be better to make sure that this port can work well on you local Mac machine first.

**Comment by longhuan2018** [comment]
> If it supports osx, you can remove `qscintilla:x64-osx` from ci.baseline.txt.
> Then try to fix the regressions.
> It might be better to make sure that this port can work well on you local Mac machine first.

I compiled qscintilla under osx successfully. But CI failed to compile under osx, but I found that it has not been compiled to qscintilla port. The failure was caused by the glib port. Is there a problem with CI's osx environment?

**Comment by NancyLi1013** [comment]
Hi @longhuan2018 
Could you please resolve the conflicts first?

**Comment by longhuan2018** [comment]
> Hi @longhuan2018
> Could you please resolve the conflicts first?

I have resolved the conflict, and now it can be compiled under osx and linux.

**Comment by NancyLi1013** [comment]
LGTM. Thanks for your update.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**